crates/sidecar/src/cache/mod.rs

```
@@ -205,21 +231,38 @@ impl DatabaseRef for Cache {
         address: Address,
         index: StorageKey,
     ) -> Result<StorageValue, Self::Error> {
-        self.iter_synced_sources()
-            .find_map(|source| {
-                let res = source.storage_ref(address, index);
-                if let Err(e) = res.as_ref() {
+        let mut saw_miss = false;
+        for source in self.iter_synced_sources() {
+            match source.storage_ref(address, index) {
+                Ok(value) => return Ok(value),
+                Err(SourceError::CacheMiss) => {
+                    saw_miss = true;
+                    debug!(
+                        target = "cache::storage_ref",
+                        name = %source.name(),
+                        address = %address,
+                        index = %index,
+                        "Cache source reported storage cache miss",
+                    );
+                }
+                Err(e) => {
                     error!(
                         target = "cache::storage_ref",
                         name = %source.name(),
                         address = %address,
                         index = %index,
                         error = ?e,
-                        "Failed to fetch the storage from cache source");
+                        "Failed to fetch the storage from cache source",
+                    );
                 }
-                res.ok()
-            })
-            .ok_or(CacheError::NoCacheSourceAvailable)
+            }
+        }
+
+        if saw_miss {
+            Ok(StorageValue::ZERO)
+        } else {
+            Err(CacheError::NoCacheSourceAvailable)
+        }
```

but if we miss all the cache sources, we should return an error, no? returning a `StorageValue` which we don't know if it is true may bring us troubles.